src/main/java/bio/terra/cda/app/util/NestedColumn.java

```diff
@@ -22,20 +25,20 @@ public static NestedColumn generate(String qualifiedColumnName) {
      */
     public static NestedColumn generate(String qualifiedColumnName)
         throws IllegalArgumentException {
-      StringBuilder unnestClause = new StringBuilder();
+      Set<String> unnestClauses = new LinkedHashSet<String>();
```

Comment:
Does this need to be a `LinkedHashSet()` to preserve the insertion order? If not I would use `HashSet`.

Reply:
I wanted to preserve the order if only for the Junit assertions of the unwind when there is more than one unnest statement generated will be done in a predictable way.

Reply:
If the only need for order is for the test, I think the test code should be changed to assert equality on the set, rather than fixing the order of the set. The test could use `assertThat()` - `assertThat(result.getUnnestClauses(), equalTo(expected));`

Reply:
Turns out it is necessary.   When unnesting the ResearchSubject as _ResearchSubject needs to defined before using _ResearchSubject.identifier in the clause.

Reply:
Makes sense